app/components/base/Button.tsx

```diff
@@ -139,11 +127,56 @@ const StyledLink = styled.a`
   display: inline-block;
 `;
 
-const ButtonLink = ({ text, size, variant, link }: Props) => (
+type Props = {
+  text: string;
+  size: "tiny" | "small" | "standard" | "large" | "block";
+  variant:
+    | "primary"
+    | "secondary"
+    | "hollow-light"
+    | "hollow-dark"
+    | "hollow-white";
+  type?: "button" | "submit" | "reset";
+  link?: { href: string; external?: boolean };
+  isDisabled?: boolean;
+  onButtonClick?: any;
+};
+
+const Button = ({
+  text,
+  size,
+  variant,
+  type = "button",
+  link,
+  isDisabled,
+  onButtonClick,
+}: Props) =>
+  link && link.href ? (
+    <ButtonLink text={text} size={size} variant={variant} link={link} />
+  ) : (
+    <StyledButton
+      type={type}
+      className={`${variant} ${size} ${isDisabled ? "disabled" : ""}`}
+      disabled={isDisabled}
+      onClick={onButtonClick}
+    >
+      {text}
+    </StyledButton>
+  );
+
+const ButtonLink = ({
+  text,
+  size,
+  variant,
+  link,
+  isDisabled,
+  onButtonClick,
+}: Props) => (
   <StyledLink
     href={link.href}
     target={link.external && "_blank"}
-    className={`${variant} ${size}`}
+    className={`${variant} ${size} ${isDisabled ? "disabled" : ""}`}
+    onClick={onButtonClick}
```

Comment:
Applying button.disabled style conditinoally. 
Adding onClick handler for button